--- origsrc/clang-14.0.5.src/tools/libclang/CMakeLists.txt	2022-06-10 22:47:36.000000000 +0800
+++ src/clang-14.0.5.src/tools/libclang/CMakeLists.txt	2023-12-14 21:05:26.005031900 +0800
@@ -2,7 +2,7 @@
 # ABI, and when it is updated, it should be updated to the current
 # LLVM_VERSION_MAJOR.
 # Please also see clang/tools/libclang/libclang.map
-set(CLANG_SONAME 13)
+set(CLANG_SONAME ${LLVM_VERSION_MAJOR})
 
 set(SOURCES
   ARCMigrate.cpp
