--- origsrc/cfe-3.7.1.src/lib/Basic/Targets.cpp	2015-11-16 15:05:53.000000000 -0600
+++ src/cfe-3.7.1.src/lib/Basic/Targets.cpp	2016-02-10 16:11:56.539778600 -0600
@@ -3765,6 +3765,7 @@ public:
     Builder.defineMacro("_X86_");
     Builder.defineMacro("__CYGWIN__");
     Builder.defineMacro("__CYGWIN32__");
+    Builder.defineMacro("__declspec(a)", "__attribute__((a))");
     addCygMingDefines(Opts, Builder);
     DefineStd(Builder, "unix", Opts);
     if (Opts.CPlusPlus)
