--- origsrc/cfe-3.7.1.src/lib/Headers/CMakeLists.txt	2015-07-30 17:47:41.000000000 -0500
+++ src/cfe-3.7.1.src/lib/Headers/CMakeLists.txt	2016-02-10 16:49:05.892870500 -0600
@@ -69,7 +69,13 @@ set(files
   xtestintrin.h
   )
 
+if(CLANG_RESOURCE_DIR)
+set(output_dir ${LLVM_RUNTIME_OUTPUT_INTDIR}/${CLANG_RESOURCE_DIR}/include)
+set(install_dir bin/${CLANG_RESOURCE_DIR}/include)
+else()
 set(output_dir ${LLVM_LIBRARY_OUTPUT_INTDIR}/clang/${CLANG_VERSION}/include)
+set(install_dir lib${LLVM_LIBDIR_SUFFIX}/clang/${CLANG_VERSION}/include)
+endif()
 
 # Generate arm_neon.h
 clang_tablegen(arm_neon.h -gen-arm-neon
@@ -98,4 +104,4 @@ set_target_properties(clang-headers PROP
 install(
   FILES ${files} ${CMAKE_CURRENT_BINARY_DIR}/arm_neon.h
   PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ
-  DESTINATION lib${LLVM_LIBDIR_SUFFIX}/clang/${CLANG_VERSION}/include)
+  DESTINATION ${install_dir})
