--- origsrc/clang-9.0.1.src/lib/Headers/CMakeLists.txt
+++ src/clang-9.0.1.src/lib/Headers/CMakeLists.txt
@@ -142,7 +142,13 @@
   openmp_wrappers/__clang_openmp_math_declares.h
 )
 
+if(CLANG_RESOURCE_DIR)
+set(output_dir ${LLVM_RUNTIME_OUTPUT_INTDIR}/${CLANG_RESOURCE_DIR}/include)
+set(install_dir bin/${CLANG_RESOURCE_DIR}/include)
+else()
 set(output_dir ${LLVM_LIBRARY_OUTPUT_INTDIR}/clang/${CLANG_VERSION}/include)
+set(install_dir lib${LLVM_LIBDIR_SUFFIX}/clang/${CLANG_VERSION}/include)
+endif()
 set(out_files)
 set(generated_files)
 

