--- origsrc/cfe-3.8.0.src/lib/Basic/Targets.cpp	2016-02-19 11:40:14.000000000 -0600
+++ src/cfe-3.8.0.src/lib/Basic/Targets.cpp	2016-06-06 03:54:52.635984200 -0500
@@ -3867,6 +3867,7 @@ public:
     Builder.defineMacro("_X86_");
     Builder.defineMacro("__CYGWIN__");
     Builder.defineMacro("__CYGWIN32__");
+    Builder.defineMacro("__declspec(a)", "__attribute__((a))");
     addCygMingDefines(Opts, Builder);
     DefineStd(Builder, "unix", Opts);
     if (Opts.CPlusPlus)
@@ -4135,7 +4136,6 @@ class CygwinX86_64TargetInfo : public X8
 public:
   CygwinX86_64TargetInfo(const llvm::Triple &Triple)
       : X86_64TargetInfo(Triple) {
-    TLSSupported = false;
     WCharType = UnsignedShort;
   }
   void getTargetDefines(const LangOptions &Opts,
@@ -4144,6 +4144,7 @@ public:
     Builder.defineMacro("__x86_64__");
     Builder.defineMacro("__CYGWIN__");
     Builder.defineMacro("__CYGWIN64__");
+    Builder.defineMacro("__declspec(a)", "__attribute__((a))");
     addCygMingDefines(Opts, Builder);
     DefineStd(Builder, "unix", Opts);
     if (Opts.CPlusPlus)
@@ -4153,6 +4154,9 @@ public:
     if (!Opts.SjLjExceptions)
       Builder.defineMacro("__SEH__");
   }
+  BuiltinVaListKind getBuiltinVaListKind() const override {
+    return TargetInfo::CharPtrBuiltinVaList;
+  }
 };
 
 class DarwinX86_64TargetInfo : public DarwinTargetInfo<X86_64TargetInfo> {
