--- origsrc/clang-9.0.1.src/cmake/modules/AddClang.cmake
+++ src/clang-9.0.1.src/cmake/modules/AddClang.cmake
@@ -82,9 +82,9 @@
       )
   endif()
 
-  if(ARG_SHARED AND ARG_STATIC)
+  if((ARG_SHARED OR BUILD_SHARED_LIBS) AND ARG_STATIC)
     set(LIBTYPE SHARED STATIC)
-  elseif(ARG_SHARED)
+  elseif(ARG_SHARED OR BUILD_SHARED_LIBS)
     set(LIBTYPE SHARED)
   else()
     # llvm_add_library ignores BUILD_SHARED_LIBS if STATIC is explicitly set,
@@ -93,6 +93,7 @@
     set_property(GLOBAL APPEND PROPERTY CLANG_STATIC_LIBS ${name})
   endif()
   llvm_add_library(${name} ${LIBTYPE} ${ARG_UNPARSED_ARGUMENTS} ${srcs})
+  set_target_properties(${name} PROPERTIES SOVERSION "${CLANG_VERSION_MAJOR}")
 
   if(TARGET ${name})
     target_link_libraries(${name} INTERFACE ${LLVM_COMMON_LIBS})
