--- origsrc/cfe-5.0.1.src/lib/Driver/ToolChains/MinGW.cpp	2017-08-10 20:47:32.000000000 -0500
+++ src/cfe-5.0.1.src/lib/Driver/ToolChains/MinGW.cpp	2017-12-31 18:18:34.610637300 -0600
@@ -365,6 +365,10 @@ bool toolchains::MinGW::UseSEHExceptions
   return getArch() == llvm::Triple::x86_64;
 }
 
+bool toolchains::MinGW::UseSjLjExceptions(const ArgList &Args) const {
+  return getArch() == llvm::Triple::x86;
+}
+
 void toolchains::MinGW::AddCudaIncludeArgs(const ArgList &DriverArgs,
                                            ArgStringList &CC1Args) const {
   CudaInstallation.AddCudaIncludeArgs(DriverArgs, CC1Args);
--- origsrc/cfe-5.0.1.src/lib/Driver/ToolChains/MinGW.h	2017-08-10 20:47:32.000000000 -0500
+++ src/cfe-5.0.1.src/lib/Driver/ToolChains/MinGW.h	2017-12-31 18:17:44.224176400 -0600
@@ -65,6 +65,7 @@ public:
   bool isPIEDefault() const override;
   bool isPICDefaultForced() const override;
   bool UseSEHExceptions() const;
+  bool UseSjLjExceptions(const llvm::opt::ArgList &Args) const override;
 
   void
   AddClangSystemIncludeArgs(const llvm::opt::ArgList &DriverArgs,
