--- origsrc/cfe-7.0.1.src/lib/Driver/ToolChains/MinGW.cpp	2019-05-06 18:41:21.210702100 -0400
+++ src/cfe-7.0.1.src/lib/Driver/ToolChains/MinGW.cpp	2019-05-06 18:42:01.116958000 -0400
@@ -516,3 +516,18 @@ void toolchains::MinGW::AddClangCXXStdli
     break;
   }
 }
+
+void toolchains::MinGW::AddCXXStdlibLibArgs(const ArgList &Args,
+                                    ArgStringList &CmdArgs) const {
+  switch (GetCXXStdlibType(Args)) {
+  case ToolChain::CST_Libcxx:
+    CmdArgs.push_back("-lc++");
+    CmdArgs.push_back("-lc++abi");
+    break;
+  case ToolChain::CST_Libstdcxx:
+    CmdArgs.push_back("-lstdc++");
+    if (Args.hasArg(options::OPT_static))
+      CmdArgs.push_back("-lpthread");
+    break;
+  }
+}
--- origsrc/cfe-7.0.1.src/lib/Driver/ToolChains/MinGW.h	2018-04-18 04:47:26.000000000 -0400
+++ src/cfe-7.0.1.src/lib/Driver/ToolChains/MinGW.h	2019-05-06 18:42:18.032527400 -0400
@@ -74,6 +74,8 @@ public:
   void AddClangCXXStdlibIncludeArgs(
       const llvm::opt::ArgList &DriverArgs,
       llvm::opt::ArgStringList &CC1Args) const override;
+  void AddCXXStdlibLibArgs(const llvm::opt::ArgList &DriverArgs,
+                           llvm::opt::ArgStringList &CC1Args) const override;
 
   void AddCudaIncludeArgs(const llvm::opt::ArgList &DriverArgs,
                           llvm::opt::ArgStringList &CC1Args) const override;
